src/podofo/main/PdfEncoding.cpp

```diff
@@ -612,13 +615,17 @@ void PdfEncoding::writeCIDMapping(PdfObject& cmapObj, const PdfFont& font, const
             range.LastCode.WriteHexTo(temp);
             output.Write(temp);
         }
+
+        output.Write("\nendcodespacerange\n");
     }
     else
     {
+        output.Write("1 begincodespacerange\n");
+
         m_Encoding->AppendCodeSpaceRange(output, temp);
-    }
 
-    output.Write("\nendcodespacerange\n");
+        output.Write("\nendcodespacerange\n");
```

Comment:
Similary you already put the "footer" of the code space range directly in `AppendCodeSpaceRange(output, temp)` and I guess this should be removed as well.

Reply:
Yes, it's a mistake. I've removed that line of code.